exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/SelectTests.kt

```diff
@@ -211,6 +213,55 @@ class SelectTests : DatabaseTestsBase() {
         }
     }
 
+    val testDBsSupportingArrays =
+        listOf(TestDB.POSTGRESQL, TestDB.POSTGRESQLNG, TestDB.H2, TestDB.H2_MYSQL, TestDB.H2_MARIADB, TestDB.H2_PSQL, TestDB.H2_ORACLE, TestDB.H2_SQLSERVER)
+
+    // adapted from `testInList01`
+    // TODO all the other tests of `inList` can be adapted to test both `inList` and `eqAny` if necessary
+    fun testEqAny(eqOp: Column<String>.() -> Op<Boolean>) {
+        withDb(testDBsSupportingArrays) {
+            withCitiesAndUsers { _, users, _ ->
+                val r = users.select { users.id.eqOp() }.orderBy(users.name).toList()
+
+                assertEquals(2, r.size)
+                assertEquals("Alex", r[0][users.name])
+                assertEquals("Andrey", r[1][users.name])
+            }
```

Comment:
I appreciate that this helps reduce redundancy, especially since you implemented the feature using both `Op` and `Function`. After refactoring, please use this (or whatever code you're testing) directly in each unit test instead. That way it will be more clear what the DSL we're testing actually looks like when used.

The tests that would be great to see are:
1. `eq anyFrom`
2. One other comparison op: the example below is good -> `greaterEq anyFrom`
3. `neq anyFrom`
4. `eq anyFrom(emptyArray())`